File: MirrorBall/app/src/main/java/com/sandystudios/mirrorball/MainActivity.java
package com.sandystudios.mirrorball;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;
import androidx.viewpager.widget.ViewPager;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.sandystudios.mirrorball.adapter.CustomPagerAdapter;

public class MainActivity extends AppCompatActivity {

    int[] mResources = {R.drawable.boxing, R.drawable.badminton, R.drawable.kohli};

    ViewPager mViewPager;
    private CustomPagerAdapter mAdapter;
    private LinearLayout pager_indicator;
    private int dotsCount;
    private ImageView[] dots;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button btn = findViewById(R.id.entry_point);
        btn.setOnClickListener(v -> {
            Intent intent = new Intent(this, LivePreviewActivity.class);
            startActivity(intent);
        });

        mViewPager = findViewById(R.id.viewpager);
        pager_indicator = (LinearLayout) findViewById(R.id.viewPagerCountDots);
        mAdapter = new CustomPagerAdapter(this, mResources);
        mViewPager.setAdapter(mAdapter);
        mViewPager.setCurrentItem(0);
        mViewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

            }

            @Override
            public void onPageSelected(int position) {
                for (int i = 0; i < dotsCount; i++) {
                    dots[i].setImageDrawable(ContextCompat.getDrawable(getApplicationContext(), R.drawable.nonselecteditem_dot));
                }
                dots[position].setImageDrawable(ContextCompat.getDrawable(getApplicationContext(), R.drawable.selecteditem_dot));
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });

        setPageViewIndicator();
    }

    private void setPageViewIndicator() {
        dotsCount = mAdapter.getCount();
        dots = new ImageView[dotsCount];

        for (int i = 0; i < dotsCount; i++) {
            dots[i] = new ImageView(this);
            dots[i].setImageDrawable(ContextCompat.getDrawable(getApplicationContext(), R.drawable.nonselecteditem_dot));

            LinearLayout.LayoutParams params = new LinearLayout.LayoutParams(
                    LinearLayout.LayoutParams.WRAP_CONTENT,
                    LinearLayout.LayoutParams.WRAP_CONTENT
            );

            params.setMargins(6, 0, 6, 0);

            final int presentPosition = i;
            dots[presentPosition].setOnTouchListener((v, event) -> {
                mViewPager.setCurrentItem(presentPosition);
                return true;
            });
            pager_indicator.addView(dots[i], params);
        }

        dots[0].setImageDrawable(ContextCompat.getDrawable(getApplicationContext(), R.drawable.selecteditem_dot));
    }
}

File: PoseDetection/app/src/main/java/com/sandystudios/posedetection/MainActivity.kt
package com.sandystudios.posedetection

import android.content.ContentValues.TAG
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.MotionEvent
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import androidx.core.content.ContextCompat
import androidx.viewpager.widget.ViewPager
import androidx.viewpager.widget.ViewPager.OnPageChangeListener
import com.sandystudios.posedetection.adapter.CustomPagerAdapter

class MainActivity : AppCompatActivity() {

    private var mResources = intArrayOf(R.drawable.boxing, R.drawable.kohli, R.drawable.badminton)

    private lateinit var mViewPager: ViewPager
    private var mAdapter: CustomPagerAdapter? = null
    private var pagerIndicator: LinearLayout? = null
    private var dotsCount = 0
    private lateinit var dots: MutableList<ImageView>

//    // creating object of ViewPager
//    private var mViewPager: ViewPager? = null
//
//    // images array
//    private var images = intArrayOf(R.drawable.boxing, R.drawable.kohli, R.drawable.badminton)
//
//    // Creating Object of ViewPagerAdapter
//    private var mViewPagerAdapter: ViewPagerAdapter? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        // Initializing the ViewPager Object
//        mViewPager = findViewById<View>(R.id.viewPagerMain) as ViewPager
//
//        // Initializing the ViewPagerAdapter
//        mViewPagerAdapter = ViewPagerAdapter(this@MainActivity, images)
//
//        // Adding the Adapter to the ViewPager
//        mViewPager!!.adapter = mViewPagerAdapter

        mViewPager = findViewById(R.id.viewpager)
        pagerIndicator = findViewById<View>(R.id.viewPagerCountDots) as LinearLayout
        mAdapter = CustomPagerAdapter(this, mResources)
        mViewPager.adapter = mAdapter
        mViewPager.currentItem = 0
        mViewPager.addOnPageChangeListener(object : OnPageChangeListener {
            override fun onPageScrolled(
                position: Int,
                positionOffset: Float,
                positionOffsetPixels: Int
            ) {
            }

            override fun onPageSelected(position: Int) {
                for (i in 0 until dotsCount) {
                    dots[i].setImageDrawable(
                        ContextCompat.getDrawable(
                            applicationContext,
                            R.drawable.nonselecteditem_dot
                        )
                    )
                }
                dots[position].setImageDrawable(
                    ContextCompat.getDrawable(
                        applicationContext,
                        R.drawable.selecteditem_dot
                    )
                )
            }

            override fun onPageScrollStateChanged(state: Int) {}
        })

        setPageViewIndicator()
    }

    private fun setPageViewIndicator() {
        dotsCount = mAdapter!!.count
        dots = ArrayList()
        for (i in 0 until dotsCount) {
            Log.e(TAG, "setPageViewIndicator: $i")
            dots[i] = ImageView(this)
            dots[i].setImageDrawable(
                ContextCompat.getDrawable(
                    applicationContext,
                    R.drawable.nonselecteditem_dot
                )
            )
            val params = LinearLayout.LayoutParams(
                LinearLayout.LayoutParams.WRAP_CONTENT,
                LinearLayout.LayoutParams.WRAP_CONTENT
            )
            params.setMargins(6, 0, 6, 0)
            dots[i].setOnTouchListener(View.OnTouchListener { v, event ->
                mViewPager!!.currentItem = i
                true
            })
            pagerIndicator?.addView(dots[i], params)
        }
        dots[0].setImageDrawable(
            ContextCompat.getDrawable(
                applicationContext,
                R.drawable.selecteditem_dot
            )
        )
    }
}